geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/FeatureSpecificReader.scala

```diff
@@ -96,9 +94,12 @@ class FeatureSpecificReader(oldType: SimpleFeatureType, newType: SimpleFeatureTy
   lazy val v1fieldreaders = buildFieldReaders(Version1Deserializer)
   lazy val v2fieldreaders = buildFieldReaders(Version2Deserializer)
 
-  def read(reuse: AvroSimpleFeature, in: Decoder): AvroSimpleFeature = {
-    // Read the version first and choose the proper deserializer
+  def defaultRead(reuse: AvroSimpleFeature, in: Decoder): AvroSimpleFeature = {
+    // read and store the version
     val serializationVersion = in.readInt()
+    AvroSimpleFeatureDecodingsCache.getAbstractReader.version = serializationVersion
```

Comment:
this seems like a bad idea to update a static variable and then reference it later. Also, I believe technically the soft reference could get invalidated, and the version could get lost. Is there a better way you could expose this in the DatumReader class?